x-pack/plugins/apm/server/routes/services/get_services/get_service_alerts.ts

```diff
@@ -37,23 +41,28 @@ export async function getServicesAlerts({
   maxNumServices = MAX_NUMBER_OF_SERVICES,
   serviceGroup,
   serviceName,
+  start,
+  end,
 }: {
   apmAlertsClient: ApmAlertsClient;
   kuery?: string;
   maxNumServices?: number;
   serviceGroup?: ServiceGroup | null;
   serviceName?: string;
+  start: number;
+  end: number;
 }) {
   const params = {
     size: 0,
     query: {
       bool: {
         filter: [
-          { term: { [ALERT_RULE_PRODUCER]: 'apm' } },
-          { term: { [ALERT_STATUS]: ALERT_STATUS_ACTIVE } },
+          ...termQuery(ALERT_RULE_PRODUCER, 'apm'),
+          ...termQuery(ALERT_STATUS, ALERT_STATUS_ACTIVE),
+          ...rangeQuery(start, end),
           ...kqlQuery(kuery),
           ...serviceGroupQuery(serviceGroup),
-          ...(serviceName ? [{ term: { [SERVICE_NAME]: serviceName } }] : []),
+          ...termQuery(SERVICE_NAME, serviceName),
```

Comment:
I remember there was a discussion about `environment` but I can't recall why we didn't filter the `environment` when fetching the alert count for the service(s)

On the other hand, when we display the alerts table on the service overview page we pass the environment filter. 

IMHO, this can be confusing. I believe we need to pass the environment if it's available (**not** in case of `All | Not defined` ) in order to have consistent data

Unless I miss something here 🤔 

any thoughts? @sqren @yngrdyn

Reply:
We decided to show alert counts across all transaction types. I think we do want to filter by environment though.

Reply:
Ok, I'll update the endpoint!